tests/unit/customizations/cloudformation/test_yamlhelper.py

```diff
@@ -94,6 +103,70 @@ def test_parse_json_with_tabs(self):
         output = yaml_parse(template)
         self.assertEqual(output, {'foo': 'bar'})
 
+    def test_parse_json_preserve_elements_order(self):
+        input_template = """
+        {
+            "B_Resource": {
+                "Key2": {
+                    "Name": "name2"
+                },
+                "Key1": {
+                    "Name": "name1"
+                }
+            },
+            "A_Resource": {
+                "Key2": {
+                    "Name": "name2"
+                },
+                "Key1": {
+                    "Name": "name1"
+                }
+            }
+        }
+        """
+        expected_dict = OrderedDict([
+            ('B_Resource', OrderedDict([('Key2', {'Name': 'name2'}), ('Key1', {'Name': 'name1'})])),
+            ('A_Resource', OrderedDict([('Key2', {'Name': 'name2'}), ('Key1', {'Name': 'name1'})]))
+        ])
+        output_dict = yaml_parse(input_template)
+        self.assertEqual(expected_dict, output_dict)
+
+    def test_parse_yaml_preserve_elements_order(self):
+        input_template = """
+        B_Resource:
+            Key2:
+                Name: name2
+            Key1:
+                Name: name1
+        A_Resource:
+            Key2:
+                Name: name2
+            Key1:
+                Name: name1
+        """
+        output_dict = yaml_parse(input_template)
+        expected_dict = OrderedDict([
+            ('B_Resource', OrderedDict([('Key2', {'Name': 'name2'}), ('Key1', {'Name': 'name1'})])),
+            ('A_Resource', OrderedDict([('Key2', {'Name': 'name2'}), ('Key1', {'Name': 'name1'})]))
+        ])
+        self.assertEqual(expected_dict, output_dict)
+
+        output_template = yaml_dump(output_dict)
+        # yaml dump changes indentation, remove spaces and new line characters to just compare the text
+        self.assertEqual(re.sub(r'\n|\s', '', input_template),
+                         re.sub(r'\n|\s', '', output_template))
+
+    def test_yaml_merge_tag(self):
```

Comment:
Not sure I understand what this test has to do with this particular change, can you elaborate?

Reply:
I could be wrong but looks like from the contributor's comment above, this test is to make sure the 'merge' tag parsing is unaffected by this change